gateway/src/main/java/io/zeebe/gateway/ClusterClient.java

```diff
@@ -15,9 +15,14 @@
  */
 package io.zeebe.gateway;
 
+import io.zeebe.gateway.api.commands.DeployWorkflowCommandStep1.DeployWorkflowCommandBuilderStep2;
 import io.zeebe.gateway.api.commands.Topology;
+import io.zeebe.gateway.api.events.DeploymentEvent;
+import io.zeebe.gateway.protocol.GatewayOuterClass.DeployWorkflowRequest;
 import io.zeebe.gateway.protocol.GatewayOuterClass.HealthRequest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.WorkflowRequestObject;
 import io.zeebe.util.sched.future.ActorFuture;
+import java.util.List;
 
 public class ClusterClient {
```

Comment:
Is this the endpoint for grpc?

Reply:
No, this is ClusterClient. As I mentioned before here at gateway we are trying to isolate things as much as possible to make testings easier (due to mocking). This is one of the tricks we used to make testing easier and way nicer than it was initially.

The point of the ClusterClient is to contact the cluster and execute some command which comes from EndpointManager.